pkg/controllers/mcs/endpointslice_controller.go

```diff
@@ -71,6 +71,19 @@ func (c *EndpointSliceController) Reconcile(ctx context.Context, req controllerr
 		return controllerruntime.Result{}, nil
 	}
 
+	// TBD: The work is managed by service_export_controller.go and endpointslice_collect_controller.go now,
+	// after the ServiceExport is deleted, the corresponding works' labels will be cleaned, so we should delete the EndpointSlice in control plane.
+	// Once the conflict between service_export_controller.go and endpointslice_collect_controller.go is fixed, the following code should be deleted.
+	if serviceName := util.GetLabelValue(work.Labels, util.ServiceNameLabel); serviceName == "" {
+		err := helper.DeleteEndpointSlice(c.Client, labels.Set{
+			workv1alpha1.WorkNamespaceLabel: req.Namespace,
+			workv1alpha1.WorkNameLabel:      req.Name,
+		})
+		if err == nil {
+			return controllerruntime.Result{}, nil
```

Comment:
if err != nil, shall we add some logs?

Reply:
The function `DeleteEndpointSlice` prints out the error log.